autofill/autofill-impl/src/main/java/com/duckduckgo/autofill/impl/ui/credential/management/viewing/AutofillManagementCredentialsMode.kt

```diff
@@ -351,16 +353,17 @@ class AutofillManagementCredentialsMode : DuckDuckGoFragment(R.layout.fragment_a
     private fun loadDomainFavicon(credentials: LoginCredentials) {
         lifecycleScope.launch {
             credentials.domain?.let {
+                val size = resources.getDimensionPixelSize(dimen.toolbarIconSize)
                 getActionBar()?.setLogo(
                     BitmapDrawable(
                         resources,
                         faviconManager.loadFromDiskWithParams(
                             tabId = null,
                             url = it,
-                            width = resources.getDimensionPixelSize(dimen.toolbarIconSize),
-                            height = resources.getDimensionPixelSize(dimen.toolbarIconSize),
+                            width = size,
+                            height = size,
                             cornerRadius = resources.getDimensionPixelSize(dimen.keyline_0),
-                        ),
+                        ) ?: faviconManager.generateDefaultFavicon(it.extractDomain().orEmpty()).toBitmap(size, size),
```

Comment:
`faviconManager.generateDefaultFavicon` will be called on the main thread. haven't looked at the details of it, but if it could be slightly heavyweight would be best to switch to another dispatcher before calling.

Reply:
Good catch! Changed to io.